packages/git/src/browser/git-widget.ts

```diff
@@ -193,10 +193,8 @@ export class GitWidget extends VirtualWidget {
                 title: 'Discard Changes',
                 onclick: async event => {
                     const repo = await this.gitRepositoryProvider.getSelected();
-                    this.git.checkout(repo, {
-                        branch: this.status.branch,
-                        paths: change.uri
-                    });
+                    const options: Git.Options.Checkout.WorkingTreeFile = { paths: change.uri };
```

Comment:
Why do I not need to specify the branch from which I want to checkout? AFAIU the HEAD will be detached when I am in a conflicting rebase situation.

Reply:
Then please use `treeish`. AFAIK, `branch` is a `treeish`.

```
git checkout [-f|--ours|--theirs|-m|--conflict=<style>] [<tree-ish>] [--] <paths>…​
git checkout [-p|--patch] [<tree-ish>] [--] [<paths>…​]
```